### DISCUSSION TITLE
Adding new concepts to checkpoint using Lora

### DISCUSSION THREAD

**Discussion by FunnyFinger** [bmaltais/kohya_ss#701] [General]
Hello,
I have this idea which I'm not sure if it is feasible or not, why can't we merge several Loras into one universal Checkpoint that contains all the concepts available in the Lora format ? For example, we might have a regular checkpoint that is not able to precisely create a transparent skin, cyborg outfit and cyclops. We could train 3 different Loras using those corresponding keywords and then merge them into one checkpoint, such that one keyword will trigger the corresponding effect, and if we use the 3 keywords, it will trigger the 3 effects simultaneously without weight distribution and degrading the effects. Such process if feasible will make us able to merge all unusual concepts into one universal checkpoint.

**Comment by noxiouscardiumdimidium** [comment]
READ THESE: do your traing on SD v1-5, to have the most "vanilla" base to train on. Train lora type: LycorisLocon

https://www.kombitz.com/2023/03/29/how-to-use-lycoris-locon-loha-models-with-automatic1111s-stable-diffusion-web-ui/
https://github.com/KohakuBlueleaf/LyCORIS

here's a cut and paste of a previous response on slightly different question, but it still applies. you can train as many subjects/concepts in the Locon as you want if youre settings/tokens are right - either as a single batch of multiple folders, or across multiple smaller batrches of individual folders, and final Locon will still know all the subject matter (as long settings and token good) this will also help you avoid lost work/time from crashes, by getting the "resume state" function to work in an actually useful way : 
                                              
                                                                       ///copypasta///
//The closest thing i've up with is an exploit to pull your trained weights after crash/ctrl+c so you can stop, test lora, then continue training if you want.

Step1. configure all settings for new project and save config.json
step2. enable "save training state" but NO RESUME FOLDER - a working folder doesn't exist yet
step3. TRAIN A SINGLE IMAGE FOR A SINGLE STEP - when it finishes a few seconds later it will create a folder named
"[whatever the project name is]-state" inside your "model" folder
step4. copy path to new "state" folder and paste into the "resume from saved training state" field in advanced settings and save config.json
step5. train as normal

IF and when you crash, or choose to stop with ctrl+c, clicking train will cause you to resume from your last completed step of training. when you would like to finish early and receive a working lora to test, perform step 6-7

step6. while training is stopped ,remove ALL images from image folder EXCEPT ONE
step7. TRAIN A SINGLE IMAGE FOR A SINGLE STEP - when it completes a few seconds later, you will receive a lora based on the total training data contained in the "[projectname]-state" folder

///DONT BOTHER TRYING THIS WITH FULL DREAMBOOTH 2GB MODLES - MY LAST UPDATE BROKE ALL KINDS OF STUFF, INCLUDING RESUME STATES FOR THE "DREAMBOOTH" TAB ; works with "standard" loras, havent tested with DyLora, Locon,Loha, LyCoris-Locon, LyCoris-Loha since i got eveerything working again

//Update: small-batch quick-and-dirty testing, SEEMS to work with LycorisLocon just fine, 100% failure rate for "Koyha DyLoRA" cant even get through 1 step to create folder with "save training state" enabled

**Reply by imranmu** [reply]
Seeing this discussion anyone know what is limiting to inject multiple LoRas into single checkpoint. I finetune multiple concepts (wine glass vs beer glass) in single LoRa but they do not work at the same time. If i use two keywords together, it doesn't work.
Is it possible to do this in Dreambooth or LycorisLocon is the only option.
